src/Microsoft.Data/StringColumn.cs

```diff
@@ -45,6 +45,17 @@ public StringColumn(string name, IEnumerable<string> values) : base(name, 0, typ
         private long _nullCount;
         public override long NullCount => _nullCount;
 
+        public override void Resize(long length)
+        {
+            if (length < Length)
+                throw new ArgumentException(Strings.CannotResizeDown, nameof(length));
+
+            for (long i = Length; i < length; i++)
+            {
+                Append(null);
+            }
+        }
+
         public void Append(string value)
```

Comment:
Implementation of `HashColumnValues` for StringColumn is coming in the next patch